### PR TITLE
chore (pkg/scale) scale integration into genesis

### PR DESCRIPTION
**Pull request by jimjbrettj**
## Changes

- Integrate new scale pkg into lib/genesis
- Replace AccountInfo type and integrate scale pkg whereever it is used
- Updates to encoding for Uint128

## Tests

```
go test ./lib/genesis -v

go test ./lib/runtime/wasmer -v

go test ./dot/rpc/modules -v
```

## Issues

- #1621

### CODE REVIEW

File: pkg/scale/comparison_test.go [hunk 1]
```diff
@@ -16,160 +16,161 @@
...
-		for _, tt := range allTests {
-			// dst := reflect.New(reflect.TypeOf(tt.in)).Elem().Interface()
-			if _, err := Marshal(tt.in); (err != nil) != tt.wantErr {
-				b.Errorf("decodeState.unmarshal() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-		}
-	}
-}
+//
```
Comment (noot) [inline pkg/scale/comparison_test.go:19]: why is this commented out?

**Review by noot — APPROVED** [review]
looks good, please uncomment commented out tests?

File: pkg/scale/comparison_test.go (hunk 1 above)
Comment (jimjbrettj) [reply pkg/scale/comparison_test.go:19]: @timwu20 Do you have suggestions on how to remove circular dependency this creates?

**Review by EclesioMeloJunior — APPROVED** [review]

File: lib/genesis/helpers.go [hunk 2]
```diff
@@ -212,7 +213,7 @@ func buildRawArrayInterface(a []interface{}, kv *keyValue) {
 			kv.iVal = append(kv.iVal, tba)
 		case float64:
 			// TODO: determine how to handle this error
-			encVal, _ := scale.Encode(uint64(v2))
+			encVal, _ := scale.Marshal(uint64(v2))
```
Comment (timwu20) [inline lib/genesis/helpers.go:216]: really should handle this error

File: pkg/scale/comparison_test.go (hunk 1 above)
Comment (timwu20) [reply pkg/scale/comparison_test.go:19]: You can remove this file for now

**Review by timwu20 — DISMISSED** [review]

File: lib/genesis/helpers.go (hunk 2 above)
Comment (jimjbrettj) [reply lib/genesis/helpers.go]: @timwu20 I handled it with a panic, but can refactor to return an error if that is prefered

Comment (timwu20) [reply lib/genesis/helpers.go]: That is preferred.  We should never panic in the codebase.

**Pull request merged**